src/socket.rs

```diff
@@ -660,6 +662,76 @@ impl Socket {
     ) -> io::Result<usize> {
         sys::send_to_vectored(self.as_raw(), bufs, addr, flags)
     }
+
+    /// Sends data on the socket to the connected peer accompanied by ancillary
+    /// control message data.
+    #[cfg(all(unix, not(target_os = "redox")))]
+    pub fn send_msg(
+        &self,
+        bufs: &[IoSlice<'_>],
+        cmsg: &CmsgWriter<'_>,
+        flags: c_int,
+    ) -> io::Result<usize> {
+        sys::sendmsg(
+            self.as_raw(),
+            std::ptr::null(),
+            0,
+            bufs,
+            cmsg.io_slice(),
+            flags,
+        )
+    }
+
+    /// Sends data on the socket to the given address accompanied by ancillary
+    /// control message data.
+    #[cfg(all(unix, not(target_os = "redox")))]
+    pub fn send_msg_to(
+        &self,
+        addr: &SockAddr,
+        bufs: &[IoSlice<'_>],
+        cmsg: &CmsgWriter<'_>,
+        flags: c_int,
+    ) -> io::Result<usize> {
+        sys::sendmsg(
+            self.as_raw(),
+            addr.as_storage_ptr(),
+            addr.len(),
+            bufs,
+            cmsg.io_slice(),
+            flags,
+        )
+    }
+
+    /// Receives data on the socket accompanied by ancillary control message data.
+    #[cfg(all(unix, not(target_os = "redox")))]
+    pub fn recv_msg(
+        &self,
+        bufs: &mut [MaybeUninitSlice<'_>],
+        control_data: &mut [MaybeUninit<u8>],
+        flags: c_int,
+    ) -> io::Result<(usize, SockAddr, CmsgIter<'_>, RecvFlags)> {
```

Comment:
Can you match the other recv functions `(usize, CmsgIter<'_>, RecvFlags, SockAddr)`?